src/NewMarkownHelpCommand.cs

```diff
@@ -0,0 +1,219 @@
+
+        [Parameter(ParameterSetName = "FromMaml")]
+        public string? ModuleGuid { get; set; }
+
+        [Parameter(ParameterSetName = "FromMaml")]
+        public string? ModuleName { get; set; }
+
+        [Parameter()]
+        public SwitchParameter NoMetadata { get; set; }
+
+        [Parameter(ParameterSetName = "FromCommand")]
+        public string? OnlineVersionUrl { get; set; }
+
+        [Parameter(Mandatory = true)]
+        public string? OutputFolder { get; set; }
+
+        [Parameter(ParameterSetName = "FromModule")]
+        [Parameter(ParameterSetName = "FromMaml")]
+        public SwitchParameter WithModulePage { get; set; }
+
+        [Parameter(ParameterSetName = "FromMaml")]
+        public SwitchParameter ConvertNotesToList { get; set; }
+
+        [Parameter(ParameterSetName = "FromMaml")]
+        public SwitchParameter ConvertDoubleDashLists { get; set; }
+
+        [Parameter()]
+        public SwitchParameter AlphabeticParamsOrder { get; set; } = true;
+
+        [Parameter()]
+        public SwitchParameter UseFullTypeName { get; set; }
+
+        [Parameter(ParameterSetName = "FromModule")]
+        [Parameter(ParameterSetName = "FromCommand")]
+        public PSSession? Session { get; set; }
+
+        [Parameter(ParameterSetName = "FromModule")]
+        [Parameter(ParameterSetName = "FromMaml")]
+        public string? ModulePagePath { get; set; }
+
+        public SwitchParameter ExcludeDontShow { get; set; }
+
+        #endregion
+
+        protected override void BeginProcessing()
+        {
+            if (Metadata is not null && NoMetadata)
+            {
+                var exception = new InvalidOperationException(Microsoft_PowerShell_PlatyPS_Resources.NoMetadataAndMetadata);
+                ErrorRecord err = new ErrorRecord(exception, "NoMetadataAndMetadata", ErrorCategory.InvalidOperation, Metadata);
+                ThrowTerminatingError(err);
+            }
+        }
+
+        protected override void ProcessRecord()
+        {
+            base.ProcessRecord();
+        }
+
+        protected override void EndProcessing()
+        {
+            if (OutputFolder is not null && !Directory.Exists(OutputFolder))
+            {
+                Directory.CreateDirectory(OutputFolder);
+            }
+
+            List<FileInfo> writtentFileList = new();
+
+            Collection<CommandHelp>? cmdHelpObjs = null;
+
+            TransformSettings transformSettings = new TransformSettings
+            {
+                AlphabeticParamsOrder = AlphabeticParamsOrder,
+                CreateModulePage = WithModulePage,
+                DoubleDashList = ConvertDoubleDashLists,
+                ExcludeDontShow = ExcludeDontShow,
+                FwLink = FwLink,
+                HelpVersion = HelpVersion,
+                Locale = Locale is not null ? new CultureInfo(Locale) : CultureInfo.GetCultureInfo("en-US"),
+                ModuleGuid = ModuleGuid is not null ? Guid.Parse(ModuleGuid) : null,
```

Comment:
This does not seem right.. it evaluates to: if ModuleGuid is null assign the value Guid.Parse(ModuleGuid). Since ModuleGuid is null, Parse will throw.

Reply:
Ah, sorry. You can keep it as is, but `x is null ? a : b` is more readable thank `x is not null ? b : a`, this also applies to the `ocale is not null` line above.